### DISCUSSION TITLE
Allow extending of CpModel class

### DISCUSSION THREAD

**Discussion by gregy4** [google/or-tools#3583] [Feature requests] [answered]
**What language and solver does this apply to?**
Java

**Describe the problem you are trying to solve.**
Preserve compatibility between or-tools versions for longer time than supported by or-tools team. Create syntactic sugar around api that suits me better.

**Describe the solution you'd like**
Remove final keyword from CpModel class.

**Describe alternatives you've considered**
Inheritance of all classes creating a model from one class with access to CpModel instance.

**Comment by gregy4** [comment, marked as answer]
It seems to me that in C# is inheritance of CpModel class not limited - no sealed keyword. In case it is true, final class in Java is an error for me.

**Comment by lperron** [comment]
Actually, it violates one of our internal design rules.
